modules/core/src/test/java/org/apache/ignite/internal/GridDiscoveryManagerChangeCoordinatorTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal;
+
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+import java.util.UUID;
+import java.util.regex.Pattern;
+import org.junit.Test;
+
+/**
+ * Tests change coordinator event logging.
+ */
+public class GridDiscoveryManagerChangeCoordinatorTest extends GridCommonAbstractTest {
+    /** Listener log messages. */
+    private final ListeningTestLogger listeningLog = new ListeningTestLogger(false, log);
+
+    /** */
+    private static final String CRD_CHANGE_MSG  = "Coordinator changed \\[prev=TcpDiscoveryNode " +
+        "\\[id=%s.*cur=TcpDiscoveryNode \\[id=%s.*";
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        listeningLog.clearListeners();
+
+        stopAllGrids();
+
+        super.afterTest();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super
+            .getConfiguration(igniteInstanceName)
+            .setGridLogger(listeningLog)
+            .setClientMode("client".equals(igniteInstanceName))
+            .setDaemon("daemon".equals(igniteInstanceName));
+    }
+
+    /**
+     * Checks that there are no messages like "coordinator changed" after client left topology.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testChangeCoordinatorLogging() throws Exception {
+        //Start 2 server nodes
+        IgniteEx srv1 = startGrid("server1");
+        startGrid("server2");
+
```

Comment:
hope would be a little bit simpler to start client between 2 servers run ? s1 -> c1 -> s2 isn`t it ?